This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marius Kittler <mariuskittler@gmx.de>
Date: Sun, 8 Nov 2020 17:10:15 +0100
Subject: [PATCH 1/2] windeployqt: Fix name of qt6CoreName variable

Pick-to: 6.0 6.0.0
Change-Id: I3b16ec0b13008abc8f38dac63456db20f2f989c5
Reviewed-by: Kai Koehne <kai.koehne@qt.io>

diff --git a/src/windeployqt/main.cpp b/src/windeployqt/main.cpp
index 1111111..2222222 100644
--- a/src/windeployqt/main.cpp
+++ b/src/windeployqt/main.cpp
@@ -1447,7 +1447,7 @@ static DeployResult deploy(const Options &options,
         }
 
         if (options.patchQt  && !options.dryRun) {
-            const QString qt5CoreName = QFileInfo(libraryPath(libraryLocation, "Qt6Core", qtLibInfix,
+            const QString qt6CoreName = QFileInfo(libraryPath(libraryLocation, "Qt6Core", qtLibInfix,
                                                               options.platform, result.isDebug)).fileName();
 #ifndef QT_RELOCATABLE
             if (!patchQtCore(targetPath + QLatin1Char('/') + qt6CoreName, errorMessage)) {

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joerg Bornemann <joerg.bornemann@qt.io>
Date: Wed, 16 Dec 2020 10:23:25 +0100
Subject: [PATCH 2/2] CMake: Fix static build of designer

The QT_DESIGNER_STATIC define was not set due to erroneous conditions.

Pick-to: 6.0
Fixes: QTBUG-89515
Change-Id: Ib3292bbc6a97346a9368598f98f9e865400d300d
Reviewed-by: Friedemann Kleint <Friedemann.Kleint@qt.io>

diff --git a/src/designer/src/components/lib/CMakeLists.txt b/src/designer/src/components/lib/CMakeLists.txt
index 1111111..2222222 100644
--- a/src/designer/src/components/lib/CMakeLists.txt
+++ b/src/designer/src/components/lib/CMakeLists.txt
@@ -878,12 +878,9 @@ qt_internal_add_resource(DesignerComponents "widgetbox"
 ## Scopes:
 #####################################################################
 
-qt_internal_extend_target(DesignerComponents CONDITION static
+qt_internal_extend_target(DesignerComponents CONDITION NOT QT_BUILD_SHARED_LIBS
     DEFINES
         QT_DESIGNER_STATIC
-)
-
-qt_internal_extend_target(DesignerComponents CONDITION NOT QT_BUILD_SHARED_LIBS
     INCLUDE_DIRECTORIES
         ../../../../shared/findwidget
         ../../../../shared/qtgradienteditor
diff --git a/src/designer/src/designer/CMakeLists.txt b/src/designer/src/designer/CMakeLists.txt
index 1111111..2222222 100644
--- a/src/designer/src/designer/CMakeLists.txt
+++ b/src/designer/src/designer/CMakeLists.txt
@@ -104,7 +104,7 @@ qt_internal_extend_target(designer CONDITION TARGET Qt::PrintSupport
         Qt::PrintSupport
 )
 
-qt_internal_extend_target(designer CONDITION QT_CONFIG___contains___static
+qt_internal_extend_target(designer CONDITION NOT QT_BUILD_SHARED_LIBS
     DEFINES
         QT_DESIGNER_STATIC
 )
diff --git a/src/designer/src/lib/CMakeLists.txt b/src/designer/src/lib/CMakeLists.txt
index 1111111..2222222 100644
--- a/src/designer/src/lib/CMakeLists.txt
+++ b/src/designer/src/lib/CMakeLists.txt
@@ -453,12 +453,9 @@ qt_internal_extend_target(Designer CONDITION TARGET Qt::OpenGLWidgets
         Qt::OpenGLWidgets
 )
 
-qt_internal_extend_target(Designer CONDITION static
+qt_internal_extend_target(Designer CONDITION NOT QT_BUILD_SHARED_LIBS
     DEFINES
         QT_DESIGNER_STATIC
-)
-
-qt_internal_extend_target(Designer CONDITION NOT QT_BUILD_SHARED_LIBS
     SOURCES
         ../../../shared/qtpropertybrowser/qtbuttonpropertybrowser.cpp ../../../shared/qtpropertybrowser/qtbuttonpropertybrowser.h
         ../../../shared/qtpropertybrowser/qteditorfactory.cpp ../../../shared/qtpropertybrowser/qteditorfactory.h
